contentScript.js

```diff
@@ -1,13 +1,19 @@
 // contentScript.js
 
 chrome.storage.local.get(["crnList"], function(result) {
-    if (result.crnList === undefined) return;
+    if (result.crnList === undefined || result.crnList.length == 0) return;
 
-    if(location.href.indexOf("bwskfreg.P_AltPin") != -1 && result.crnList) {
-        let crnInputs = document.querySelectorAll('.dataentrytable input[name="CRN_IN"]');
-        
-        for(let i = 0; i < result.crnList.length && i < crnInputs.length; i++) {
-            crnInputs[i].value = result.crnList[i];
+    if(location.href.indexOf("classRegistration") != -1 && result.crnList) {
+        const crns = document.querySelector("#crns");
+        const addCRNbutton = document.querySelector("#addCRNbutton");
+        crns.innerHTML = "";
+
+        for(let i = 0; i < result.crnList.length; i++) {
+            const newInput = document.createElement("input");
+            newInput.value = result.crnList[i];
+            crns.appendChild(newInput);
         }
+        addCRNbutton.click();
```

Comment:
Could we reconsider the current approach of modifying the form's DOM to append new input fields? 
It seems we're only clicking the `addCRNbutton.click();` only once after all self-crafted fields have been added, which may not be the most reliable method.

What I envision for this functionality is:

- **Eliminate direct `DOM` manipulation** (No `appendChild` use) to avoid potential conflicts with the underlying `AngularJS` on the course registration page. We're not fully aware of the existing `Design By Contract` of the CityU IT codebase, and forcing changes to the DOM via methods like `appendChild` might violate it.
- **Trigger click on the `#addCRNbutton` only as it’s required.** For instance, if the user provides only 3 CRNs, we do not need to simulate any button clicks. 

> Conversely, if a user provides 7 CRNs, the browser extension should operate as follows:
> 1. Input the first 5 CRNs.
> 2. Execute `addCRNbutton.click();`
> 3. Enter the 6th CRN.
> 4. Execute `addCRNbutton.click();`
> 5. Enter the 7th CRN.
> 

Adopting this more natural interaction pattern ensures we **align with the CityU system's existing flow**, reducing the risk of unexpected issues with the extension and maintaining the integrity of the platform's expected operations. This process will give our extension the appearance of a user-driven sequence, making it more robust and less susceptible to interruptions due to unforeseen system behaviour.

Reply:
I agree. Will change accordingly.